feat: add streaming arg to createApp() (#15483)
withastro/astro: .changeset/warm-donuts-learn.md

---
'astro': minor
---

> Adds `streaming` option to the `createApp()` function exported from `astro/app/entrypoint`

Reviewer: ```suggestion
Adds a new `streaming` option to the Adapter API to allow third-party adapters to disable HTML streaming
```
So, maybe something like this? (whatever the exact *purpose* is; I'm assuming that it's to allow them to set it false if they want?) The first line is easier for people to figure out if it applies to them if it references  *adapters* specifically, in some way (e.g. "the Adapter API", "adapter authors", "third-party adapters") 

And when people are looking to read what's changed, I generally prefer that the first line:
- if there *is* a new thing they didn't know about before, specifically name check it so that they know what to search for in docs if they want to learn more (i.e. `streaming`) 
- but *don't* make this an API reference type of statement (used in this function, exported from this module...)
- instead, focus on its purpose/value (in addition to making clear if it's for only a particular subset of our audience)

> `createApp()` allows adapters to create an `App` instance that works in development and in production.

Reviewer: This feels to me like not necessary information for adapter authors? Wouldn't they already know/be doing this? (And again, if we go with a more functional opening line, then here is where we can instead give the more technical details if you want)

```suggestion
```

> It now accepts `streaming` (defaults to `true`) as an option, forwarded to the `App` constructor:

Reviewer: ```suggestion
An adapter's `createApp()` function now accepts `streaming` (defaults to `true`) as an option. HTML streaming breaks a document into chunks to send over the network and render on the page in order. This normally results in visitors seeing your HTML as fast as possible but factors such as network conditions and waiting for data fetches can block page rendering.

Adapter authors may want to disable streaming in the case of/to allow.... However, this may result in....

To disable HTML streaming for your adapter, pass `streaming: false` when you create your `App` instance:
```
Maybe something like this (not needing the one-liner paragraph before this)?

---
'astro': minor
---

Adds `streaming` option to the `createApp()` function in the Adapter API, mirroring the same functionality available when creating a new `App` instance

> 

Author: ```suggestion
```
